SavingsManagementSystem.Service/User/Implementations/AdminService.cs

```diff
@@ -35,5 +44,41 @@ public async Task<RegistrationResponse> RegisterAsync(RegistrationRequest reques
 
 			return response;
 		}
+
+
+		public async Task SendMemberInviteAsync(string email)
+		{
+			//create a verification token for the link
+			var vToken = await _vTokenService.CreateVerificationTokenAsync(30, "Successful", email);
+			if (vToken == null)
+			{
+				throw new InvalidOperationException("Operation Failed...");
+			}
+			var encodedToken = TokenConverter.EncodeToken(vToken.Token);
+			await _unit.SaveChangesAsync();
+
+			// Load the email template from the file
+			var htmlPath = Path.Combine("StaticFiles", "Html", "MemberInvite.html");
+			var emailTemplate = File.ReadAllText(htmlPath);
+			var queryParams = $"email={email}&token={encodedToken}";
+			var inviteLink = LinkGenerator.GenerateUrl("VerifyLink", "Auth", queryParams);
+
+			// Replacing the {{INVITE_LINK}} placeholder with the actual reset link
+			emailTemplate = emailTemplate.Replace("{{INVITE_LINK}}", inviteLink);
+			var mailRequest = new MailRequest()
+			{
+				Subject = "Member Invite",
+				RecipientEmail = email,
+				Body = emailTemplate
+			};
+			try{ await _mailService.SendEmailAsync(mailRequest);}
```

Comment:
in the try, do the same thing in catch block, but the status is the difference

Reply:
i have already created the status to be successful at the top, so if eventually, the email doesn't send, it is going to change to the status to Fail.
			var vToken = await _vTokenService.CreateVerificationTokenAsync(30, "Successful", email);

Reply:
No, you only update it to successful if it is successful